### PR TITLE
[JIT] add unused and is_scripting to docs

### PR DESCRIPTION
**Pull request by eellison**

### CODE REVIEW

**Review by suo — APPROVED** [review]

File: docs/source/jit.rst [hunk 1]
```diff
@@ -216,7 +217,7 @@ lazily compiled in the order they are used in ``forward``, as well as any
 
 Functions
 ~~~~~~~~~
-Functions don't change much, they can be decorated with :func:`@torch.jit.ignore <torch.jit.ignore>` if needed.
+Functions don't change much, they can be decorated with :func:`@torch.jit.ignore <torch.jit.ignore>` or `torch.jit.unused <torch.jit.unused>` if needed.
```
Comment (driazati) [inline docs/source/jit.rst]: Use the `:func:` directive so it will generate a link to the function's docs

File: docs/source/jit.rst [hunk 2]
```diff
@@ -183,7 +183,8 @@ The new usage looks like this:
 
 * The module's ``forward`` is compiled by default. Methods called from ``forward`` are lazily compiled in the order they are used in ``forward``.
 * To compile a method other than ``forward`` that is not called from ``forward``, add ``@torch.jit.export``.
-* To stop the compiler from compiling a method and leave it as a call to Python, add ``@torch.jit.ignore``.
+* To stop the compiler from compiling a method, add ``@torch.jit.ignore`` or `@torch.jit.unused`. ``@ignore`` leaves the
```
Comment (driazati) [inline docs/source/jit.rst]: Use `:func:` on all of these here too

File: docs/source/jit.rst [hunk 3]
```diff
@@ -1097,6 +1105,10 @@ check the correctness of the model as you go.
 
 .. autofunction:: ignore
 
+.. autofunction:: unused
```
Comment (driazati) [inline docs/source/jit.rst]: The code example in docstring for `unused`  should have a `.. testcode::` directive on it, see the other documented functions for examples

**Review by driazati — COMMENTED** [review]
Some small issues but they should be resolved before merging

See https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#building-documentation for how to build / view the docs

File: docs/source/jit.rst (hunk 3 above)
Comment (eellison) [reply docs/source/jit.rst]: it already does

Comment (driazati) [reply docs/source/jit.rst]: Sorry I meant `is_scripting` above, `unused` looks fine

**Pull request closed without merging**